snuba/datasets/tags_column_processor.py

```diff
@@ -16,6 +27,39 @@
 NESTED_COL_EXPR_RE = re.compile(r"^([a-zA-Z0-9_\.]+)\[([a-zA-Z0-9_\.:-]+)\]$")
 
 
+@dataclass(frozen=True)
+class ParsedNestedColumn:
+    """
+    Provides some structure to the tags/contexts column to avoid parsing the string
+    in multiple places around the code.
+    """
+
+    col_name: str
+    parameter: Optional[str]
```

Comment:
We really need to come up with a common vocabulary around these concepts, it took me a while to figure out what "parameter" was referring to in this context.